### ISSUE TITLE
[BUG] Using Dropzone with an interactive element inside causes accessibility violation

### ISSUE THREAD

**Issue by emilyuhde** [react-dropzone/react-dropzone#1152]
**Describe the bug**
If you use the Dropzone component with an interactive element inside of it, such as a button used in the "[Opening File Dialog Programmatically]()" example, it will cause an a11y issue because you have nested interactive elements. In the case of this example, it's a `<div role="button">` around a `<button>` but the same will be true if you use a link instead.

**To Reproduce**
1. The react-dropzone doc site and go to the [Opening File Dialog Programmatically]() example.
2. Right click the "Open File Dialog" button in the example and inspect the element.

Alternatively, you can run Axe (https://www.deque.com/axe/browser-extensions/) on the same example.

**Expected behavior**
There are no nested interactive elements and no accessibility violations.

**Screenshots**
<img width="1366" alt="Screen Shot 2022-03-08 at 3 08 56 PM" src="https://user-images.githubusercontent.com/4451556/157317579-614053ab-f0c9-4ab2-a16e-43d1bce5c5b9.png">
<img width="1424" alt="Screen Shot 2022-03-08 at 3 17 34 PM" src="https://user-images.githubusercontent.com/4451556/157317965-b9b71870-ed0a-4485-8885-0a9e381f6cd2.png">

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Chrome, but it'll be the same in any browser
 - Version: 99.0.4844.51

**Comment by rolandjitsu** [comment]
This was probably introduced by https://github.com/react-dropzone/react-dropzone/commit/207282b801596059d938c418f21291a00ba3c917. I wonder which is the most widely used scenario so that we can decide if we remove the default `button` role or keep it.

**Comment by emilyuhde** [comment]
A possible solution would be adding a valid override to this example, if having the default role of "button" is the most common case.

`<div {...getRootProps({role: 'presentation'})}>`

**Comment by Crevitus** [comment]
@emilyuhde Thanks for this! I added `<div {...getRootProps({role: 'none'})}>` (a synonym for presentation) as it had more semantic meaning. This fixed it for me in the case where I have a nested button like in the original issue. 

The docs should probably be updated to use one of these roles in the nested button example.

**Comment by rolandjitsu** [comment]
@emilyuhde @Crevitus I agree. I think describing the default and having an example to override makes sense. I'll try to add one the upcoming weekend.

**Comment by vpfaulkner** [comment]
For what it's worth, I'm using Bootstrap and `role="button"` is causing the cursor to have the `pointer` style for the entire dropzone (which is most of the page in my case).

`getRootProps({ role: 'none' })}` fixes it for me but I'd expect others to run into something similar given how popular Bootstrap is.

**Comment by rolandjitsu** [comment]
It sounds to me like using `button` as default may not have been a good idea. I think using `presentation` might make more sense.

**Comment by theSdev** [comment]
I've been browsing the issues for a report on the accessibility issues, and was very surprised to see the `role` was changed to `presentation` here *on accessibility grounds*! In my book `<div role="presentation" tabindex="0">`, which is the HTML for the basic example on the website, is one of the bigger accessibility crimes possible. How does making a meaningless `div` intractable with `tabindex`, and then marking it decorative using `role="presentation"` makes any sense accessibility-wise?

IMO, the solution to the original issue is: Don't put any interactive elements inside this already interactive element! Sure, you're passing the check by changing the `role` to `presentation`, but the outer `div` is still very much interactive, as the `tabindex` suggests, so this change is just a bad workaround.
